packages/AI/Providers/Vectors-Pinecone/src/models/PineconeDatabase.ts

```diff
@@ -186,17 +186,19 @@ export class PineconeDatabase extends VectorDBBase {
             response.Duplicates = [];
             for(const match of queryResponse.data.matches){
                 const record: {id: string, score: number, metadata: any} = match;
-                if(!record.metadata || !record.metadata.PrimaryKeys){
+                if(!record || !record.id){
                     continue;
                 }
 
-                const metadata: {PrimaryKeys: string[]} = record.metadata;
-                if(!metadata.PrimaryKeys){
+                //splitID should include 3 elements: the entity ID, entity document ID, and the record ID
+                const splitID: string[] = record.id.split('_');
+                if(splitID.length !== 3){
+                    LogError(`Invalid vector record ID: ${record.id}`);
                     continue;
                 }
 
                 let duplicate: PotentialDuplicate = new PotentialDuplicate();
-                duplicate.LoadFromList(metadata.PrimaryKeys);
+                duplicate.KeyValuePairs.push({ FieldName: 'ID', Value: splitID[2] });
                 duplicate.ProbabilityScore = record.score;
```

Comment:
@js-bc I'm confused why there is code related to duplicate data in the PineconeDatabase class - that class is specific to a given provider, what does this have to do with the generic idea of duplicate detection?

Reply:
You're right. I've moved the function out of the VectorDBBase class and put it inside the DuplicateRecordDetector class